fix(frontend): Agent Tools & Metadata not available for V1 conversations

kub938 opened the pull request:
## Summary of PR
- Fixed an issue where the Agent Tools & Metadata list and modal were not displayed in V1.
- Updated systemMessage handling to accommodate both V0 and V1 message types.
- Defined the SystemMessageForModal interface and added ChatCompletionToolParam[] type for tools.
- In V1, the appropriate data for openhands_version and agent_class was not found, so they are currently set to null. I would appreciate guidance on how to improve this.

## Change Type
- [v] Bug fix

## Checklist
- [v] I have read and reviewed the code and I understand what the code is doing.
- [v] I have tested the code to the best of my ability and ensured it works as expected.

## Fixes
Resolves #12169 

## Release Notes
- [ ] Include this change in the Release Notes.

hieptl reviewed (commented):
Hello @kub938, thank you for submitting this pull request. Could you please add tests to cover your changes? Thank you! 🙏

kub938 commented:
# Changes Made

## 1. SystemMessage UI Rendering & Interaction Tests
- Verify Button Rendering: Confirmed that the "Show Agent Tools" button is correctly rendered within the ToolsContextMenu.
- Data Display Validation: Verified that the adapted data is accurately displayed in the SystemMessageModal.
- Integration Flow: Confirmed that clicking the "Show Agent Tools" button successfully triggers the modal and displays the expected adapted content.

## 2. adaptSystemMessage Logic Tests
- Content Mapping: Verified that the function correctly transforms input data and outputs the intended content.
- Edge Case Handling: Confirmed that the function returns null as expected when an empty array ([]) is provided as input.

## 3. Extracted adaptSystemMessage logic
- Extracted adaptSystemMessage logic into src/utils/system-message-adapter.ts. This separates business logic from UI concerns and allows for independent unit testing.

amanape commented on frontend/src/utils/system-message-adapter.ts:
```diff
@@ -0,0 +1,48 @@
+  events: OHEvent[],
+): SystemMessageForModal | null {
+  let systemMessage: SystemMessageForModal | null = null;
+  // 여기의 events타입은 2가지가 들어올 수 있다. v0와 v1타입
```
Please translate any comments to English

amanape reviewed (changes requested):
Thank you, I’ve simplified and separated your test files. I just have a few requests:

kub938 commented on frontend/__tests__/components/context-menu/tools-context-menu.test.tsx:
```diff
@@ -0,0 +1,71 @@
+import { ToolsContextMenu } from "#/components/features/controls/tools-context-menu";
+
+// 기존의 로직이 v0 를 기준으로 짜여져 있고 v1 타입에 대한 어댑팅을 추가한 것이기 때문에 v1 타입에 대한 테스트를 추가
+// 이 로직이 정상적으로 작동되면 system message 모달에서 v1 타입의 system prompt가 정상적으로 표시됨
```
Sorry about that! I’ve deleted them since they were redundant.
